tools/compute_accuracy.py

```diff
@@ -0,0 +1,240 @@
+            dtype=np.float32)
+        for ix, in_ in enumerate(inputs):
+            input_[ix] = caffe.io.resize_image(in_, self.image_dims)
+
+        if oversample:
+            # Generate center, corner, and mirrored crops.
+            input_ = caffe.io.oversample(input_, self.crop_dims)
+        else:
+            # Take center crop.
+            center = np.array(self.image_dims) / 2.0
+            crop = np.tile(center, (1, 2))[0] + np.concatenate([
+                -self.crop_dims / 2.0,
+                self.crop_dims / 2.0
+            ])
+            input_ = input_[:, crop[0]:crop[2], crop[1]:crop[3], :]
+
+        # Classify
+        caffe_in = np.zeros(np.array(input_.shape)[[0,3,1,2]],
+                            dtype=np.float32)
+        for ix, in_ in enumerate(input_):
+            caffe_in[ix] = self.transformer.preprocess(self.inputs[0], in_)
+        out = self.forward_all(**{self.inputs[0]: caffe_in})
+
+        # Where it differs from the original Caffe class
+        if out_layer_name == None:
+            predictions = out[self.outputs[-1]]
+        else:
+            predictions = out[out_layer_name]
+
+        # For oversampling, average predictions across crops.
+        if oversample:
+            predictions = predictions.reshape((len(predictions) / 10, 10, -1))
+            predictions = predictions.mean(1)
+
+        return predictions
+
+def compute_accuracy(snapshot, deploy_file, labels_file, mean_file, img_set, width, height, resize_mode, grayscale=False, oversample=False):
+    """
+    Evaluate a Net on a set of images, and dump the result in two
+    pickle files.
+    Returns True on sucess
+
+    Arguments:
+    snapshot -- a Caffe trained model
+    deploy_file -- the corresponding deploy file
+    labels_file -- the file containing the dataset labels
+    mean_file -- the dataset mean file
+    img_set -- the file containing a list of images
+    resize_mode -- the mode used to resize images
+
+    Keyword arguments:
+    oversampling -- boolean, True if oversampling should be enabled
+    """
+
+    img_matrix = pd.read_csv(img_set, engine='python', header=None,sep=r" (?!(.*) )")
+
+    mean_blob = caffe.proto.caffe_pb2.BlobProto()
+    mean_data = open(mean_file, 'rb').read()
+    mean_blob.ParseFromString(mean_data)
+    mean = np.array(caffe.io.blobproto_to_array(mean_blob))[0].mean(1).mean(1)
+
+    # Loading the classifier
+    if config_value('caffe_root')['cuda_enabled'] and\
+            config_value('gpu_list'):
+        caffe.set_mode_gpu()
+
+    if grayscale:
+        net = Classifier(deploy_file, snapshot,
+                       mean=mean,
+                       raw_scale=255)
+    else:
+        net = Classifier(deploy_file, snapshot,
+                       mean=mean,
+                       channel_swap=(2,1,0),
+                       raw_scale=255)
+
+    size = len(img_matrix)
+    num_classes = np.array(img_matrix[2])[-1]+1
```

Comment:
I don't know what's supposed to be in `img_matrix`, but this isn't right. For my dataset with 20 classes, this is returning 16. Are you assuming that the images are sorted by class?